client/flutter/lib/pages/onboarding/notifications_page.dart

```diff
@@ -25,14 +32,31 @@ class _NotificationsPageState extends State<NotificationsPage> {
   }
 
   void _allowNotifications() async {
+    // iOS needs a call for permissions but Android doesn't.
+    // If the user opts-out on Android we just don't register the device token.
+    if (Platform.isIOS) {
+      await _firebaseMessaging.requestNotificationPermissions(
```

Comment:
Does this return a value?  The user could (accidentally or intentionally) hit cancel on the dialog.

Reply:
From what Chris told me sounds like we are intended to use `onIosSettingsRegistered` to get that feedback.  I wouldn't block on this but we should restructure to set the preference based on that value rather than set it true always here.  (let's create an issue for later)